specinfra.gemspec

```diff
@@ -20,7 +20,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ["lib"]
 
   spec.add_runtime_dependency "net-scp"
-  spec.add_runtime_dependency "net-ssh", ">= 2.7", (RUBY_VERSION >= "2.0" ? "< 3.1" : "< 2.10")
+  spec.add_runtime_dependency "net-ssh", ">= 2.7", (RUBY_VERSION >= "2.0" ? "< 4.0" : "< 2.10")
```

Comment:
Could you change this line like below, please ?

```ruby
spec.add_runtime_dependency "net-ssh", ">= 2.7", "< 4.0"
```

Because gemspec is evaluated when building, not installing. So checking `RUBY_VERSION` does not work as intended.

Ref #505 https://github.com/mizzy/specinfra/pull/514#issuecomment-171959746